### PR TITLE
Run the  Consumer MessageHandler in a Task

### PR DESCRIPTION
**Pull request by Gsantomaggio**
Thanks to @Zerpet for help and to @jonnepmyra for rasing the issue and testing the pr 
Fixes https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/issues/246

Run the Consumer MessageHandler in a Task
---
The chunks are now processed in a separate [Task](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/250/files#diff-b08ff9a3d581af9df4f7e209452830c93f2367debb487e937e158870332fbd1eR263-R272)
In this way, the socket thread is free to handle other requests like `ConsumerUpdate`

Convert `DispatchMessage` method from `void` to `async Task`
---

[`DispatchMessage`](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/pull/250/files#diff-b08ff9a3d581af9df4f7e209452830c93f2367debb487e937e158870332fbd1eR195-R198) is now an `async task` by removing the `wait` the code is aync

### CODE REVIEW

**Review by lukebakken — COMMENTED** [review]
I've reviewed once but want to re-read again!

**Comment by Gsantomaggio** [comment]
Thank you, @lukebakken. I convert the pr into a draft. I am working with @jonnepmyra to validate it; it has yet to be ready.

**Comment by jonnepmyra** [comment]
Been testing this in many different scenarios the last week, and the current state of this PR is a **huge improvement** that solves multiple problems that've experienced when using SuperStreams.

Some examples of problems that's been solved:
* Deadlocks for slow consumers
* Running out of credits on multiple "single active consumer" switches
* None of the consumers where activated (== all consumers is "stuck" in a "waiting" status)
* Consumer connections that were silently dropped

We're still struggling with cases of "all consumers are stuck in waiting status", but this happends alot less frequently with this PR, and seems to correlate with heavy load in RabbitMQ (or possibly the host of the application instance) and more investigation is needed (Will continue a separate discussion with @Gsantomaggio about this, since it seems to be out of context of this PR and very hard to reproduce).

File: RabbitMQ.Stream.Client/Consts.cs [hunk 1]
```diff
@@ -8,11 +8,23 @@ namespace RabbitMQ.Stream.Client
 {
     internal static class Consts
     {
+        private static readonly Random s_random = new();
```
Comment (ricardSiliuk) [inline RabbitMQ.Stream.Client/Consts.cs:11]: I think you can drop this and just use `Random.Shared.Next()` on L22 and L27 :)

https://andrewlock.net/building-a-thread-safe-random-implementation-for-dotnet-framework/

**Pull request merged**